### DISCUSSION TITLE
feature request: accept npm package names as entrypoints

### DISCUSSION THREAD

**Discussion by threepointone** [cloudflare/workers-sdk#3473] [Ideas]
Let's say I made an application, like a session config service built with Workers, named `workers-session-service` on npm. I'd like to be able to be able to share it with folks, and let them be able to deploy it without writing any code. So, this should work: 

- `npm install workers-session-service`: installs the package
- `npx wrangler dev workers-session-service`: should spin it up locally
- `npx wrangler publish workers-session-service`: should publish it to the edge

It should also read from wrangler.toml if available. Which leads to fun questions about provisioning resources like kv automatically (which we've discussed before). Still, being able to do just the above should make distribution easier.

**Comment by penalosa** [comment]
Moving this to a discussion for now, since I think it requires more thought—especially around resource provisioning and the implications there
